listenstore/listenstore/listenstore.py

```diff
@@ -89,30 +87,48 @@ def insert_postgresql(self, listens):
-                        %(recording_msid)s, %(raw_data)s) ON CONFLICT DO NOTHING """, self.format_dict(listen))
+                    params = self.format_dict(listen)
+                    res = connection.execute(text("""
+                        INSERT INTO listen(user_id, ts, artist_msid, album_msid, recording_msid)
+                             VALUES (:user_id, to_timestamp(:ts), :artist_msid, :album_msid,
+                                    :recording_msid)
+                        ON CONFLICT DO NOTHING
+                          RETURNING id
+                    """), params)
+                    params['_id'] = res.fetchone()[0]
+
+                    res = connection.execute(text("""
+                        INSERT INTO listen_json(id, data)
+                             VALUES (:_id, :data)
+                        ON CONFLICT DO NOTHING
+                    """), params)
                 except Exception, e:     # Log errors
-                        self.log.error(e)
-
-    def execute(self, query, params={}):
-        with self.engine.connect() as connection:
-            res = connection.execute(query, params)
-            return res.fetchall()
+                    self.log.error(e)
 
     def fetch_listens_from_storage(self, user_id, from_id, to_id, limit, order, precision):
-        query = """ SELECT id, user_id, extract(epoch from ts), artist_msid, album_msid, recording_msid, raw_data """ + \
-                """ FROM listen WHERE user_id = %(user_id)s """ + \
-                """ AND extract(epoch from ts) > %(from_id)s AND extract(epoch from ts) < %(to_id)s  """ + \
-                """ ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """ LIMIT %(limit)s"""
-        params = {
-            'user_id': user_id,
-            'from_id': from_id,
-            'to_id': to_id,
-            'limit': limit
-        }
-
-        results = self.execute(query, params)
-        for row in results:
-            yield self.convert_row(row)
+        with self.engine.connect() as connection:
+            results = connection.execute(text("""
+                SELECT listen.id
+                     , user_id
+                     , extract(epoch from ts)
+                     , artist_msid
+                     , album_msid
+                     , recording_msid
+                     , data
+                  FROM listen
+                     , listen_json
+                 WHERE listen.id = listen_json.id
+                   AND user_id = :user_id
+                   AND extract(epoch from ts) > :from_id
+                   AND extract(epoch from ts) < :to_id
+              ORDER BY extract(epoch from ts) """ + ORDER_TEXT[order] + """
+                 LIMIT :limit
+            """), {
+                'user_id': user_id,
+                'from_id': from_id,
+                'to_id': to_id,
+                'limit': limit
+            })
+
+            for row in results.fetchall():
+                yield self.convert_row(row)
```

Comment:
This yield is due to the way that the cassandra library worked. We don't need it here, you can just return a list